main/main.c

```diff
@@ -0,0 +1,78 @@
+#include "discovery_beacon.h"
+#include "driver_setup.h"
+#include "esp_log.h"
+#include "http_server.h"
+#include "persistent_settings.h"
+#include "socketcand_server.h"
+
+// Name that will be used for logging
+static const char *TAG = "main";
+
+void app_main(void) {
+  // Create an event loop.
+  ESP_ERROR_CHECK(esp_event_loop_create_default());
+
+  // Initialize the networking stack.
+  ESP_ERROR_CHECK(esp_netif_init());
+
+  // Wait for hardware to start
+  vTaskDelay(pdMS_TO_TICKS(200));
+
+  // Initialize nonvolatile memory storage
+  ESP_ERROR_CHECK(persistent_settings_init());
+
+  // Set up the long-press settings reset button
+  ESP_ERROR_CHECK(persistent_settings_setup_reset_button());
+
+  // Read saved settings
+  persistent_settings_t settings;
+  ESP_ERROR_CHECK(persistent_settings_load(&settings));
+
+  // Print the settings over uart
+  char settings_json[512];
+  int bytes_written = persistent_settings_to_json(&settings, settings_json,
+                                                  sizeof(settings_json));
+  if (bytes_written == -1) {
+    ESP_LOGE(TAG, "Couldn't get JSON of persistent settings. Aborting.");
+    abort();
+  }
+  ESP_LOGI(TAG, "Current settings:\n%s\n", settings_json);
```

Comment:
I think this is going to have to allocate enough memory to fit this json string again. I think you might be able to get around it by logging settings_json directly in a separate call?

Reply:
Should be fixed in 6aff9ac.